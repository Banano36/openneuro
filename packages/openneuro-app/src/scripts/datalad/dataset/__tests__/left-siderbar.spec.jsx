import React from 'react'
import { mount } from 'enzyme'
import LeftSidebar, { SidebarRow } from '../left-sidebar.jsx'

const fixedDate = new Date('2019-04-02T19:56:41.222Z')

describe('LeftSidebar component', () => {
  it('renders with basic props', () => {
    expect(
      shallow(
        <LeftSidebar
          datasetId="ds000001"
          draftModified={fixedDate}
          snapshots={[]}
          location="/dataset/ds000001"
        />,
      ),
    ).toMatchSnapshot()
  })
  describe('SidebarRow', () => {
    it('renders draft version correctly', () => {
      const wrapper = shallow(
        <SidebarRow
          datasetId="ds000001"
          id="draft"
          draft
          modified={fixedDate}
        />,
        expect(wrapper).toMatchSnapshot(),
      )
    })
    it('renders snapshot version correctly', () => {
      const wrapper = shallow(
        <SidebarRow
          datasetId="ds000001"
          id="ds000001:1.0.0"
          modified={fixedDate}
        />,
        expect(wrapper).toMatchSnapshot(),
      )
    })
    it('toggles the active state for matching snapshot versions', () => {
      const wrapper = shallow(
        <SidebarRow
          datasetId="ds000001"
          id="ds000001:1.0.0"
          version="1.0.0"
          modified={fixedDate}
          active="1.0.0"
        />,
      )
      expect(wrapper.find('Link').hasClass('active')).toBeTruthy()
    })
    it('disables the active state for non-matching snapshot versions', () => {
      const wrapper = shallow(
        <SidebarRow
          datasetId="ds000001"
          id="ds000001:1.0.0"
          modified={fixedDate}
          active="1.0.5"
        />,
      )
      expect(wrapper.find('Link').hasClass('active')).toBeFalsy()
    })
    it('toggles the active state for matching draft versions', () => {
      const wrapper = shallow(
        <SidebarRow
          datasetId="ds000001"
          id="ds000001:1.0.0"
          modified={fixedDate}
          draft
          active="draft"
        />,
      )
      expect(wrapper.find('Link').hasClass('active')).toBeTruthy()
    })
    it('disables the active state for non-matching draft versions', () => {
      const wrapper = shallow(
        <SidebarRow
          datasetId="ds000001"
          id="ds000001:1.0.0"
          modified={fixedDate}
          draft
          active="1.0.0"
        />,
      )
      expect(wrapper.find('Link').hasClass('active')).toBeFalsy()
    })
  })
})
