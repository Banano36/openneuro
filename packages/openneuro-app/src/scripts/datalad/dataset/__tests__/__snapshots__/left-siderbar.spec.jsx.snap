// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftSidebar component SidebarRow renders draft version correctly 1`] = `
<li
  key="draft"
>
  <Link
    className=""
    to="/datasets/ds000001"
  >
    <div
      className="clearfix"
    >
      <div
        className=" col-xs-12"
      >
        <span
          className="dataset-type"
        />
        <span
          className="date-modified"
        >
          2019-04-02
        </span>
        <span
          className="icons"
        />
      </div>
    </div>
  </Link>
</li>
`;

exports[`LeftSidebar component SidebarRow renders snapshot version correctly 1`] = `
<li
  key="ds000001:1.0.0"
>
  <Link
    className="active"
    to="/datasets/ds000001/versions/undefined"
  >
    <div
      className="clearfix"
    >
      <div
        className=" col-xs-12"
      >
        <span
          className="dataset-type"
        />
        <span
          className="date-modified"
        >
          2019-04-02
        </span>
        <span
          className="icons"
        />
      </div>
    </div>
  </Link>
</li>
`;

exports[`LeftSidebar component renders with basic props 1`] = `
<div
  className="left-sidebar"
>
  <span
    className="slide"
  >
    <div
      className="snapshot-select"
      role="presentation"
    >
      <span>
        <h3>
          Versions
        </h3>
        <ul />
      </span>
    </div>
  </span>
</div>
`;
